pulp_rpm/app/shared_utils.py

```diff
@@ -1,12 +1,18 @@
 import createrepo_c as cr
+from logging import getLogger
 import tempfile
+import traceback
 import shutil
 from hashlib import sha256
+from pgpy.pgp import PGPSignature
+import rpmfile
```

Comment:
TODO(dralley): These would be significant dependencies, I need to see if they are acceptable or if there is a more suitable method

Reply:
Agreed, and I'm absolutely not tied to these deps in particular, these are just what I happened to get it working with.

Reply:
Right, totally reasonable.  Having looked at `rpmfile` it looks like basically a partial reimplementation of RPM which I'm not thrilled about, I would definitely prefer to use RPM proper since createrepo_c needs it anyway.  

I'll ping them to get the ball rolling a bit faster.

Reply:
So the feedback I got is that pulpcore and the debian plugin use `python-gnupg`, so it would be best if we can depend on that instead.

Reply:
We already chatted about this on Matrix, but I'll respond here too for posterity.

There is no equivalent in `python-gnupg` to the in-memory parsing of the signature packet that `pgpy` does. Nor does it even give access to the signer key id. However, all `python-gnupg` is is a wrapper around a subprocess call-out to `gpg`, so we could just cut out the middleman and do that ourselves if we don't mind the extra overhead of writing the signature to disk and calling gpg in a subprocess. That change would look [something like this](https://github.com/sdherr/pulp_rpm/blob/check_signer_key_nodeps/pulp_rpm/app/shared_utils.py#L173-L237). 

As for dropping the `rpmfile` dep, that's harder unless `createrepo_c` adds the python bindings [that we're asking for](https://github.com/rpm-software-management/createrepo_c/issues/346). You can definitely access those headers through the `rpm` module, *if* `python3-rpm` is installed, which is not a given. And since that's a module that wants to be installed as on OS package instead of through pip, requiring it is harder. The best solution is if `createrepo` makes these headers available since that's already installed, so we'll see what they say.

Reply:
I've looked over the code for `rpmfile` and it looks.. OK.  There's nothing terribly objectionable, just that it's overly simple for the number of edge cases that I know exist, but for the sake of a small number of tags it should probably be fine.

Reply:
I should have an answer on the package signing questions I have by the end of the week.  The people I've been trying to ask are very busy with working on DNF5 but I have a 1hr meeting with them directly soon.

Reply:
@sdherr It looks like `pgpy` does not support `EdDSA`.  I think that's probably a longer-term issue since RSA is used everywhere... it may eventually become one but I'd rather `pgpy` be a short-term solution anyway.